### ISSUE TITLE
Multiple openmp test regressions in 14.x

### ISSUE THREAD

**Issue by mgorny** [llvm/llvm-project#53732]
I'm seeing multiple test regressions in openmp in 14.0.0_rc1 compared to 13.0.1. I can reproduce them on main as well.

The tests failing for me are:
```
  libomp :: affinity/kmp-hw-subset.c
  libomp :: affinity/teams-affinity.c
  libomp :: api/omp_aligned_alloc.c
  libomp :: api/omp_aligned_calloc.c
  libomp :: atomic/kmp_atomic_cas.c
  libomp :: atomic/kmp_atomic_cas_cpt.c
  libomp :: ompt/parallel/nested_lwt_thread_num.c
  libomp :: ompt/parallel/nested_serialized_task_frames.c
  libomp :: ompt/parallel/region_in_expl_task_task_frames.c
  libomp :: ompt/tasks/explicit_task_thread_num.c
  libomp :: tasking/kmp_task_depend_all.c
  libomp :: tasking/kmp_taskwait_depend_all.c
  libomp :: tasking/kmp_taskwait_nowait.c
```

This is on Gentoo Linux amd64. I'm using sys-libs/glibc 2.34, kernel 5.16.5 if that could be relevant. The library is built using GCC 11.2.1, and the test compiler is clang 14.0.0rc1 (building using clang gives even more failures). I'm building libomp standalone against installed LLVM 14.0.0rc1.

Full log: [sys-libs:libomp-15.0.0.9999:20220210-211220.log](https://github.com/llvm/llvm-project/files/8044341/sys-libs.libomp-15.0.0.9999.20220210-211220.log)

Many of the failures seem to be related to undefined symbols. Others I can't figure out. I was trying to bisect one of them and I've simply reached the commit adding the test, so I suspect many of them are broken from day one.

I'm willing to help debugging this but I'd really use some advice what to do.

**Comment by tstellar** [comment]
This issue was not resolved in time for 14.0.0 and will need to be fixed in the next release (14.0.1).  If you disagree with this or you feel there was a mistake, please change the milestone back to 'LLVM 14.0.0 Release' so we can discuss.

**Comment by mgorny** [comment]
I'm going to file a new bug for test failures in 15.x.
